src/test/java/dao/RoleDaoImplTest.java

```diff
@@ -0,0 +1,53 @@
+package dao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+import mate.academy.dao.RoleDao;
+import mate.academy.dao.impl.RoleDaoImpl;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RoleDaoImplTest extends AbstractTest {
+    private RoleDao roleDao;
+    private Role role;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        roleDao = new RoleDaoImpl(getSessionFactory());
+        role = new Role(Role.RoleName.ADMIN);
+    }
+
+    @Test
+    public void save_Ok() {
+        Role actual = roleDao.save(role);
+        role.setId(1L);
+        assertNotNull(actual);
+        assertEquals(role, actual);
```

Comment:
```suggestion
        Role actual = roleDao.save(role);
        assertNotNull(actual);
        assertEquals(role.getName(), actual.getName());
```
setting id is a little bit weird